ndscan/plots/image_2d.py

```diff
@@ -130,11 +139,14 @@ def __init__(self, image_item: pyqtgraph.ImageItem,
         self.x_range = None
         self.y_range = None
         self.image_data = None
+        self.error_data = None
 
         #: Whether to average points with the same coordinates.
         self.averaging_enabled = False
+        #: Assumed number of samples per point for calculating the combined uncertainty.
+        self.num_samples_per_point = 1
         #: Keeps track of the running average and the number of samples therein.
-        self.averages_by_coords = dict[tuple[float, float], tuple[float, int]]()
+        self.averages_by_coords = dict[tuple[float, float], tuple[float, float, int]]()
```

Comment:
Can we make this a a frozen dataclass/… instead of a tuple? This is pretty opaque to the casual reader.